src/core/features/course/pages/list-mod-type/list-mod-type.ts

```diff
@@ -125,19 +138,46 @@ export class CoreCourseListModTypePage implements OnInit {
             const result = await CoreCourseHelper.addHandlerDataForModules(sections, this.courseId);
 
             this.sections = result.sections;
+
+            this.lastShownSectionIndex = -1;
+            this.showMoreActivities();
         } catch (error) {
             CoreDomUtils.showErrorModalDefault(error, 'Error getting data');
         }
     }
 
+    /**
+     * Show more activities.
+     *
+     * @param infiniteComplete Infinite scroll complete function. Only used from core-infinite-loading.
+     */
+    showMoreActivities(infiniteComplete?: () => void): void {
+        this.canLoadMore = false;
+
+        let modulesLoaded = 0;
+        let i: number;
+        for (i = this.lastShownSectionIndex + 1; i < this.sections.length; i++) {
+            modulesLoaded += this.sections[i].modules.length;
+
+            if (modulesLoaded >= CoreCourseListModTypePage.PAGE_LENGTH) {
+                break;
+            }
+        }
+
+        this.lastShownSectionIndex = i;
+        this.canLoadMore = i < this.sections.length;
```

Comment:
```suggestion
        let modulesLoaded = 0;

        while (this.lastShownSectionIndex < this.sections.length - 1 && modulesLoaded >= CoreCourseListModTypePage.PAGE_LENGTH) {
            this.lastShownSectionIndex++;

            modulesLoaded += this.sections[this.lastShownSectionIndex].modules.length;
        }

        this.canLoadMore = this.lastShownSectionIndex < this.sections.length - 1;
```

This simplifies the code and takes care of @dpalou's open discussion.